src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -572,18 +572,25 @@ protected function getContainerBaseClass()
      */
     protected function initializeContainer()
     {
+        $asFiles = true;
         $class = $this->getContainerClass();
         $cacheDir = $this->warmupDir ?: $this->getCacheDir();
         $cache = new ConfigCache($cacheDir.'/'.$class.'.php', $this->debug);
         $oldContainer = null;
         if ($fresh = $cache->isFresh()) {
+            if (!$asFiles) {
+                require_once $cache->getPath();
```

Comment:
moving this outside of the try/catch+require_cone where it was before means this will fail badly when the file is missing (will happen on race conditions)

Reply:
This code was present in symfony 2.X + there is not try/catch block here. So I don't get what you mean

Reply:
I moved that in the catch block